frontend/controller/actions/identity.js

```diff
@@ -146,7 +146,22 @@ export default (sbp('sbp/selectors/register', {
           }
         }
       }
-      // note: leaving groups will happen when we sync the removeOurselves message
+      // NOTE: should sync all the identity contracts which are not part of same group
+      // these users could be the ones from 12n direct messages
+      const chatRoomUsers = [...new Set(
+        Object.keys(
+          pickWhere(state.contracts, ({ type }) => type === 'gi.contracts/chatroom')
+        ).map(cID => Object.keys(state[cID].users)).flat())
+      ]
```

Comment:
@Silver-IT (Assuming it's getting rid of any possible duplicated username from an array here) Seems like this can be DRYed by using `uniq` function [here ](https://github.com/okTurtles/group-income/blob/0439e89600e5af9eae07f7a2190144cdfa1d0264/frontend/model/contracts/shared/giLodash.js#L132)in `giLodash.js`

Reply:
Yeah. We can do this way.
```
const chatRoomUsers = uniq(Object.keys(
  pickWhere(state.contracts, ({ type }) => type === 'gi.contracts/chatroom')
).map(cID => Object.keys(state[cID].users)).flat())
```